dynamic/main.go

```diff
@@ -85,7 +80,14 @@ func initialSetup() (tfbridge.ProviderInfo, pfbridge.ProviderMetadata, func() er
 			}
 
 			tfServer = p
-			defaultInfo.P.(*proto.Provider).Replace(p)
+			if tfServer != nil {
+				info = providerInfo(ctx, tfServer)
+			}
+
+			err = pfbridge.XParameterizeResetProvider(ctx, info, metadata)
```

Comment:
Is this the variable which contains the lambda? Is it passed as a reference?

Reply:
If your talking about the lambda introduced in provider.go:

https://github.com/pulumi/pulumi-terraform-bridge/blob/8c172bf6702ddd8403be87eb2b83fdc32bdebabb/pf/tfbridge/provider.go#L213-L222

Then `ctx` is the variable which contains the lambda. `XParameterizeResetProvider` unpacks the lambda from `ctx` and then calls it.

Reply:
The `metadata` used on this line comes from line 51, as far as I can see, right?

The same `metadata` also contains this function which captures it.

It looks a bit odd and I was wondering if it's passed as a reference for this to work.

Reply:
Ah, I think I know what you mean. In Go, lambdas always have access to the location of variables in their scope. If your familiar with C++, then lambdas close over the lvalue of variables they reference.